SEP/SEP-0005/sep-0005.md

```diff
@@ -92,7 +92,9 @@ WHERE
 }
 ```
 
-We will also introduce a second new keyword `ACCEPTLANG`, to refer to taking the order from the accept-lang http header provided by the client and a new shorthand for testing if a value has a matching language string `hasLang(?label, "EN")`.
+### ACCEPTLANG: a new keyword
+
+We will also introduce a second new keyword `ACCEPTLANG`, to refer to taking the order from the accept-lang http header provided by the client and a new shorthand for testing if a value has a matching language string `hasLang(?label, "EN")`. In situations where there is no http header such as comand lines, the tools should look in places such as the locale settings on Linux, or other fields where a language preference by the user of the system might be indicated.
```

Comment:
I still think this is requiring too much of implementations

Consider a distributed client-server implementation that uses a non-HTTP protocol.  Should an implementation consider the locale on the client side or the server side?

Client side locale implies that `ACCEPTLANG` works either by some form of substitution semantics i.e. the client should replace `ACCEPTLANG` with the relevant locale information otherwise any implementation is now forced to carry locale information over its transport protocol.

Server side locale leaves it more up to the implementation but you might get results that bear no relevance to the client e.g. a Russian user queries a Greek server with `ACCEPTLANG` and gets Greek results because that's the locale the server is running in.

Then there's also the case of what happens if you layer a HTTP endpoint on top of such an architecture.  What locale information are you expected to be used then?

I think the definition of this needs to be much clearer.  I would prefer an approach similar to how `default-graph` etc  is supported in the protocol layer e.g.

> `ACCEPTLANG` requires the SPARQL endpoint to substitute the requesting clients `Accept-Lang` HTTP Header identified locales into the received query in place of the keyword.  In non-HTTP contexts implementations `MAY` substitute other relevant locale information e.g. OS environment variables if appropriate but `MAY` choose to treat this keyword as specifying no additional locales.

Reply:
> `ACCEPTLANG` requires the SPARQL endpoint to substitute the requesting clients `Accept-Lang` HTTP Header identified locales into the received query in place of the keyword.

I'd be very concerned about language like this that would require protocol implementations to be able to fully parse (or at least tokenize) a query to properly support this feature. The approach taken for `default-graph` does not require this, instead using spec text to require the protocol-supplied dataset to override any dataset defined in the query string.

Reply:
@kasei Yeah I totally agree, my suggested wording is probably has just as many issues as the original wording.  I was trying to get to something like `default-graph` but probably failed